www-servers/nginx-unit/nginx-unit-1.12.0-r2.ebuild

```diff
@@ -27,23 +27,33 @@ DEPEND="perl? ( dev-lang/perl:= )
 	php7-2? ( dev-lang/php:7.2[embed] )
 	php7-3? ( dev-lang/php:7.3[embed] )
 	python? ( ${PYTHON_DEPS} )
-	ruby? ( dev-lang/ruby:= )"
+	ruby? ( dev-lang/ruby:= )
+	ssl? ( dev-libs/openssl:= )"
 RDEPEND="${DEPEND}"
 S="${WORKDIR}/${MY_P}"
 
 pkg_setup() {
 	use python && python-single-r1_pkg_setup
 }
 
+src_prepare() {
+	eapply_user
+	sed -i '/^CFLAGS/d' auto/make
+}
+
 src_configure() {
-	./configure \
-		--control=unix:/run/${PN}.sock \
-		--ld-opt="${LDFLAGS}" \
-		--log=/var/log/${PN} \
-		--modules=$(get_libdir)/${PN} \
-		--pid=/run/${PN}.pid \
-		--prefix=/usr \
-		--state=/var/lib/${PN} || die "Core configuration failed"
+	local opt=(
+		--control=unix:/run/${PN}.sock
+		--log=/var/log/${PN}
+		--modules=$(get_libdir)/${PN}
+		--pid=/run/${PN}.pid
+		--prefix=/usr
+		--state=/var/lib/${PN}
+	)
+	use ssl && opt+=( --openssl )
+	./configure ${opt[@]} --ld-opt="${LDFLAGS}" || die "Core configuration failed"
+	# For modules, add option for position-independent code
+	export CFLAGS="${CFLAGS} -fPIC"
```

Comment:
The most common way is to use `append-flags -fPIC` from toolchain-funcs.eclass.

Reply:
Ah, I knew I had come across a related eclass before, but could not remember which one.

Reply:
Actually, append-cflags from flag-o-matic.eclass looks like a better choice to me.